lua/neoclip/sorted_set.lua

```diff
@@ -47,6 +47,16 @@ local insert = function(self, entry, opts)
     end
 end
 
+local edit = function(self, entry, newEntry)
+    local key = hash(entry)
+    local new_key = hash(newEntry)
+    local node = self.entries[key];
+    node.value = newEntry -- Replace entry in node.
+
+    self.entries[key] = nil
+    self.entries[new_key] = node
```

Comment:
Don't we also need to update the underlying linked list? (`self.ll`)

Reply:
The idea behind this piece of code is to only replace the contents of the node, and not the node itself. This way, we won't change the order of the nodes.

The reason this function is that complex is because I also have to update the dictionary's reference to the nodes, using the new calculated hash for the node's content.

Reply:
Hmm, but the linked list node also holds a value (which is used by pop for example). Could maybe change but at the moment I think also the value of the linked list node need to be updated. Could you maybe also add some test case for that?

Reply:
I think I'm a bit confused.

The `self.entries` dictionary has the linked list's nodes as values (as seen [here](https://github.com/AckslD/nvim-neoclip.lua/blob/main/lua/neoclip/sorted_set.lua#L27)). So doing `self.entries[key].value` is already updating the value of the linked list's node, no?

And yeah, I like the idea of adding some test cases. I'll give it a try

Reply:
Hey @AckslD I just added some tests for this new `replace` function. Let me know what you think